### PR TITLE
Hasn/feat: button component

### PR DESCRIPTION
**Pull request by hasan-deriv**
* Created button component

### CODE REVIEW

File: src/components/ui/button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      },
+      fullwidth: {
+        true: "flex w-full",
+      },
+    },
+    compoundVariants: [
+      {
+        variant: "contained", 
+        color: "primary",
+        className: "bg-primary hover:bg-primary-hover focus:bg-primary-hover"
```
Comment (shayan-deriv) [inline src/components/ui/button/index.tsx]:
```suggestion
        className: "bg-primary hover:bg-primary-hover focus:bg-primary-hover",
```

same for couple of more in this file

Comment (shayan-deriv) [reply src/components/ui/button/index.tsx]: this is because of eslint error

Comment (hasan-deriv) [reply src/components/ui/button/index.tsx]: fixed

File: src/components/ui/button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { Slot } from '@radix-ui/react-slot';
+import { type VariantProps } from 'class-variance-authority';
+import { cn } from 'Utils/cn';
+import { buttonVariants } from './variants';
+
+export interface ButtonProps
```
Comment (matin-deriv) [inline src/components/ui/button/index.tsx]: What is the reason that you're exporting this?

Comment (hasan-deriv) [reply src/components/ui/button/index.tsx]: no reason. Shadcn/ui did it 😄 . I removed it.

**Review by matin-deriv — APPROVED** [review]

**Pull request merged**